packages/core-utils/src/io-utils.ts

```diff
@@ -1,3 +1,17 @@
+/**
+ * Attempt to parse a JSON string and return the decoded value.
+ * If the string cannot be parsed, return undefined.
+ *
+ * @param maybeJSONString - The string to attempt to parse.
+ */
+export const maybeParseJSON = (maybeJSONString: string): unknown => {
```

Comment:
Should the type of the return be `object | undefined`? I think that seems like what we can return, which might be more clear than `unknown`?

Reply:
`object` doesn't include primitive types (string, number, etc), but JSON may contain primitive values.
https://www.typescripttutorial.net/typescript-tutorial/typescript-object-type/